docs/best-practices/testing.md

```diff
@@ -0,0 +1,32 @@
+# Testing best practices
+
+## Polling
+If you need to poll to wait for some event to happen, avoid using directly the `wait.PollImmediate` function: Kass Fleet Manager provides a rich set of utilities that will handle for you all the caveats of polling when dealing with it.
+
+Such utilities are:
+* **The poller builder**. A builder interface that will guide you in configuring a custom poller
+* **Cluster pollers**. A number of preconfigured pollers to be used to poll for a specific cluster event:
+    * **WaitForClustersMatchCriteriaToBeGivenCount**. Awaits for the number of clusters with an assigned cluster id to be exactly `count`
+    * **WaitForClusterIDToBeAssigned**. Awaits for clusterID to be assigned to the designed cluster
+    * **WaitForClusterToBeDeleted**. Awaits for the specified cluster to be deleted
+    * **WaitForClusterStatus**. Awaits for the cluster to reach the desired status
+* **Kafka pollers**. A number of preconfigured pollers that will allow to easily poll for a specific kafka event:
+    * **WaitForNumberOfKafkaToBeGivenCount**. Awaits for the number of kafkas to be exactly X
+    * **WaitForKafkaCreateToBeAccepted**. Creates a kafka and awaits for the request to be accepted
+    * **WaitForKafkaToReachStatus**. Awaits for a kafka to reach a specified status
+    * **WaitForKafkaToBeDeleted**. Awaits for a kafka to be deleted
+    * **WaitForKafkaClusterIDToBeAssigned**. Await for kafka's ClusterID to be assigned.
```

Comment:
same as above - the logic changed and cluster id is assigned during creation

Reply:
Yeah, looks like this one is not used anymore.